extensions/security/runtime/src/main/java/io/quarkus/security/runtime/QuarkusIdentityProviderManagerImpl.java

```diff
@@ -127,6 +127,9 @@ public Uni<SecurityIdentity> apply(SecurityIdentity securityIdentity) {
                         return handleProvider(pos + 1, providers, request);
                     }
                 });
+        if (pos > 0) {
```

Comment:
this would be easier to understand if you extracted recursion to dedicated method and dropped `if`

Reply:
meaning: lines 114-129